src/codemagic/tools/app_store_connect.py

```diff
@@ -1195,6 +1242,10 @@ def _save_certificates(
     ) -> List[pathlib.Path]:
         return [self._save_certificate(c, private_key, p12_container_password) for c in certificates]
 
+    @staticmethod
+    def _is_device_udid_valid(device_udid: str) -> bool:
```

Comment:
I find that this method would be better suited under `Device` class. Also, could you provide a documentation reference for this UDID structure?

Reply:
Sorry, I inferred this UDID structure from the existing values, so no documentation links.
Should we remove this validation?

Reply:
This is not an official source, but from [here](https://getupdraft.com/get-my-udid) I can see that
> There are **two formats of ID**: iOS devices that came on the market between 2007 and 2018 have a 40-digit lowercase hexadecimal code and all devices after 2018 have a 25-digit uppercase hexadecimal code.

Perhaps we can omit validation on our end and just require the UDID not to be empty string, and leave all verification to Apple.